addons/Media_Recorder/src_webpack/worker/MP3ConvertHandler.jsm

```diff
@@ -0,0 +1,161 @@
+        this.worker = null;
+        this.scriptURL = null;
+        this.isValid = false;
+        this.validationTimoutID = null;
+
+        if (!this.isSupported()) {
+            console.log('Your browser doesn\'t support web workers.');
+            return;
+        }
+
+        let scriptBlob = this.createBlobWithScript();
+        this.scriptURL = URL.createObjectURL(scriptBlob);
+        this._validateScript();
+    }
+
+    isSupported() {
+        return !!window.Worker;
+    }
+
+    isWorkerExist() {
+        return !!this.worker;
+    }
+
+    _validateScript() {
+        let self = this;
+        return new Promise(resolve => {
+            this.worker = new Worker(this.scriptURL);
+            this.worker.onmessage = function(e) {
+                if (e.data === "WORKER STARTED") {
+                    self.isValid = true;
+                    resolve(true);
+                } else {
+                    console.log('Error occurred for Web worker in Media Recorder: ' + e.data);
+                    self.isValid = false;
+                    resolve(false);
+                }
+            }
+            this.worker.postMessage({
+                cmd: "validate",
+                origin: document.location.origin
+            });
+            this.validationTimoutID = setTimeout(() => {
+                if (!self.isValid) {
+                    console.log('Lib for web worker in Media Recorder is unreachable.');
+                    self.isValid = false;
+                    this.terminateWorkerProcess();
+                    resolve(false);
+                }
+            }, 5000)
+        })
+    }
+
+    execute(numberOfChannels, sampleRate, sampleLength, leftChannelData, rightChannelData) {
+        return new Promise((resolve, reject) => {
+            if (!this.isValid) {
+                reject("Not valid worker");
+            }
+            if (this.isWorkerExist()) {
+                this.terminateWorkerProcess();
+            }
+            this.worker = new Worker(this.scriptURL);
+            this.worker.onmessage = function(e) {
+                resolve(e.data);
+            }
+            this.worker.postMessage({
+                cmd: "start",
+                data: {
+                    numberOfChannels: numberOfChannels,
+                    sampleRate: sampleRate,
+                    sampleLength: sampleLength,
+                    leftChannelData: leftChannelData,
+                    rightChannelData: rightChannelData
+                },
+                origin: document.location.origin
+            });
+        })
+    }
+
+    createBlobWithScript() {
+        const script = `
+            addEventListener("message", function(e) {
+                let data = e.data;
+                if (!data.origin) {
+                    postMessage("Unknown origin");
+                    return;
+                }
+
+                const lameScriptURL = data.origin + "/media/icplayer/libs/lame.min.js";
```

Comment:
czy upewniłeś się, że to będzie działać na mC?

Reply:
Ps. Paczka zip nie była podmieniana podczas wystawiania więc należy sugerować się funkcjonalnością getState, który to powinien mieć plik mpeg-3 w stanie zamiast wav